src/site/topic24.rst

```diff
@@ -268,6 +268,130 @@ Remove Minimum & Maximum
+        } else if (child.getData().compareTo(element) > 0) {
+            return remove(element, child, child.getLeft());
+        } else {
+            return remove(element, child, child.getRight());
+        }
+    }
+
+* The private ``remove`` is basically doing a binary search through the tree looking for the value to be removed
+* Unlike the binary search however, this method must
+    * Remove the element
+    * Potentially address a gap in the tree if the node being removed is an internal node
+    * Ensure the binary search tree ordering is preserved
+
+* To do this, we make use of another private method called ``findReplacementNode``
+
+
+.. code-block:: java
+    :linenos:
+
+    private Node<T> findReplacementNode(Node<T> toRemove) {
+        Node<T> replacementNode = null;
+        if (toRemove.getLeft() == null && toRemove.getRight() == null) {
+            replacementNode = null;
+        } else if (toRemove.getLeft() != null && toRemove.getRight() == null) {
+            replacementNode = toRemove.getLeft();
+        } else if (toRemove.getLeft() == null && toRemove.getRight() != null) {
+            replacementNode = toRemove.getRight();
+        } else {
+            Node<T> parent = toRemove;
+            Node<T> child = toRemove.getRight();
+
+            // Find the in order successor (right child's left
+            // most node (minimum node))
+            while (child.getLeft() != null) {
+                parent = child;
+                child = child.getLeft();
+            }
+
+            // Set replacement node's left to
+            // the node being removed's (subtree root's) left
+            child.setLeft(toRemove.getLeft());
+
+            // If the immediate in order successor is NOT the
+            // node being replaced's right child, the parent
+            // node's new left becomes the child node's right
+            // and the child node's right is replaced with
+            // the node being replaced's right
+            if (toRemove.getRight() != child) {
+                parent.setLeft(child.getRight());
+                child.setRight(toRemove.getRight());
+            }
+            replacementNode = child;
+        }
+        return replacementNode;
+    }
+
+* ``findReplacementNode`` looks rather intimidating at first, but it you take your time looking at it, you will see that it's simple
+* The first part, the first ``if``s and ``else if``s check if the node being removed is a leaf node, or if it has one child
+    * If it's a leaf node, there is no replacement
+    * If there is only one child, then that child becomes the replacement
+
+* The other case is when there are two children
+* Here, the idea is to find the in order successor node
+    * The right child's left most node
+
+* You may see that this code is very similar to the private ``removeMax`` method
+    * Iterate down the tree until we find the left most node in the subtree
+    * This node will become the replacement node
```

Comment:
Could possibly even use that method here instead? It might make it easier to understand.